test/powershell/Language/Parser/TypeAccelerator.Tests.ps1

```diff
@@ -1,25 +1,428 @@
+                Type        = [System.Management.Automation.PSModuleInfo]
+            }
+            @{
+                Accelerator = 'powershell'
+                Type        = [System.Management.Automation.Powershell]
+            }
+            @{
+                Accelerator = 'runspacefactory'
+                Type        = [System.Management.Automation.Runspaces.RunspaceFactory]
+            }
+            @{
+                Accelerator = 'runspace'
+                Type        = [System.Management.Automation.Runspaces.Runspace]
+            }
+            @{
+                Accelerator = 'initialsessionstate'
+                Type        = [System.Management.Automation.Runspaces.InitialSessionState]
+            }
+            @{
+                Accelerator = 'psscriptmethod'
+                Type        = [System.Management.Automation.PSScriptMethod]
+            }  
+            @{
+                Accelerator = 'psscriptproperty'
+                Type        = [System.Management.Automation.PSScriptProperty]
+            }
+            @{
+                Accelerator = 'psnoteproperty'
+                Type        = [System.Management.Automation.PSNoteProperty]
+            }
+            @{
+                Accelerator = 'psaliasproperty'
+                Type        = [System.Management.Automation.PSAliasProperty]
+            }
+            @{
+                Accelerator = 'psvariableproperty'
+                Type        = [System.Management.Automation.PSVariableProperty]
+            }
+        )
+        
+        if ( $IsCoreCLR )
+        {
+            $totalAccelerators = 80 
+        } 
+        else
+        {
+            $totalAccelerators = 82
+
+            $nonCoreTypeAcceleratorTestCases = @(
+                @{
+                    Accelerator = 'adsi'
+                    Type        = [System.DirectoryServices.DirectoryEntry]
+                }
+                @{
+                    Accelerator = 'adsisearcher'
+                    Type        = [System.DirectoryServices.DirectorySearcher]
+                }
+                @{
+                    Accelerator = 'wmiclass'
+                    Type        = [System.Management.ManagementClass]
+                }
+                @{
+                    Accelerator = 'wmi'
+                    Type        = [System.Management.ManagementObject]
+                }
+                @{
+                    Accelerator = 'wmisearcher'
+                    Type        = [System.Management.ManagementObjectSearcher]
+                }
+            )
+        }
+    }
 
     It "Basic type accelerator usage" {
-        [msft_2174855] | Should Be ([int])
+        [userDefinedAcceleratorType] | Should Be ([int])
     }
 
     It "Can query type accelerators" {
-        if ( $IsCoreCLR ) { $count = 80 } else { $count = 82 }
-        $TypeAccelerators.Count -gt $count | Should Be $true
+        $TypeAccelerators.Count -gt $totalAccelerators | Should Be $true

```

Comment:
I think it would be good to test against the exact number of built-in accelerators (89). And it's recommended to write as `$TypeAccelerators.Count | Should Be $totalAccelerators` because it would give more information compared to `Should Be $true` in case the test fails.